### PR TITLE
Add home, which, and where commands

### LINKED ISSUE #40: Add home, which, and where commands
General use case: figure out what version of a package is resolved to by a name. There's no direct way of doing this without either `fetch`-ing the package and seeing what output you get, or `ls`-ing the packages and searching yourself

```
pwr which jdk
```

This would print `jdk:17.0.4` to the terminal. This would only inspect the fetched tags and would not require the package to be downloaded. Some use cases could include quickly determining the latest version of a package (as in the above example).

Another use case is printing the location of a package. I'm not sure of any way to do this currently.

```
pwr where jdk
```

This would print to the terminal the location of that package, e.g. `C:\Users\*\AppData\Roaming\pwr\pkg\jdk-17.0.4`, an error would be printed if the package resolved to by the name had not been fetched already.

Current implementation in `wh` branch.

### PR DESCRIPTION
**Pull request by tbateson**
Closes #40

### CODE REVIEW

**Review by troyaf — CHANGES_REQUESTED** [review]
Thoughts on making `where` and `which` `switch`es and re-use the `Packages` param? Looks good otherwise.

**Comment by tbateson** [comment]
> Thoughts on making `where` and `which` `switch`es and re-use the `Packages` param? Looks good otherwise.

Meh. I don't like the idea of typing `pwr -packages jdk -where` since `which`/`where` is my use case is really only designed for single package usage, whereas `-Packages` is an array. Plus it's a lot more characters to type.

**Comment by troyaf** [comment]
> > Thoughts on making `where` and `which` `switch`es and re-use the `Packages` param? Looks good otherwise.
> 
> Meh. I don't like the idea of typing `pwr -packages jdk -where` since `which`/`where` is my use case is really only designed for single package usage, whereas `-Packages` is an array. Plus it's a lot more characters to type.

I'm thinking now it makes the most sense to have which and where be `commands on the packages.

File: src/pwr.ps1 [hunk 1]
```diff
@@ -886,6 +897,17 @@ if ($Run) {
...
+} elseif ($Which) {
+	$local:Pkg = Assert-PwrPackage $Which
+	Write-PwrHost "$($local:Pkg.Ref) $($local:Pkg.Digest)"
+	exit
+} elseif ($Where) {
+	$local:Pkg = Assert-PwrPackage $Where
+	if (-not (Test-PwrPackage $local:Pkg)) {
+		Write-PwrFatal "package $($local:Pkg.Ref) is not installed"
+	}
+	Write-PwrHost (Resolve-PwrPackge $local:Pkg)
```
Comment (troyaf) [inline src/pwr.ps1:909]: Do we want to use regular write host so the pwr prefix is not in the output? I'm imagining a use where you inline the command as part of another command.

Something like

`PS > Get-ChildItem (pwr where jdk) -File -Recurse -Include '*target file*' `

**Comment by tbateson** [comment]
> I'm thinking now it makes the most sense to have which and where be `commands` on the `packages`. If you wanna assert the package length is 1, I'd be okay with that.
After thinking about this more I think changing it to be a command makes a lot more sense. I've added `home`, `where`, and `which` commands, since they all operate on packages (except for `home`).

File: src/pwr.ps1 (hunk 1 above)
Comment (tbateson) [reply src/pwr.ps1]: `Write-PwrHost` does not append the `pwr: ` prefix.

Comment (troyaf) [reply src/pwr.ps1]: Ah

**Review by troyaf — APPROVED** [review]

**Pull request merged**